bagel/mappings.py

```diff
@@ -40,35 +44,76 @@
 
 # TODO: Use importlib.resources.files(bagel) to get the path to the pipeline-catalog instead?
 PROCESSING_PIPELINE_PATH = (
-    Path(__file__).parent / "pipeline-catalog" / "processing"
+    Path(__file__).parent / "pipeline-catalog/processing/processing.json"
 )
+PROCESSING_PIPELINE_URL = "https://raw.githubusercontent.com/nipoppy/pipeline-catalog/refs/heads/main/processing/processing.json"
+
+
+def get_pipeline_catalog(get_url: str, get_path: Path) -> dict:
+    """
+    Load the pipeline catalog from the remote location or, if that fails,
+    from the local backup.
+    """
+    try:
+        response = httpx.get(get_url)
+        response.raise_for_status()
+        return response.json()
+    except (httpx.HTTPStatusError, httpx.ConnectError, json.JSONDecodeError):
+        warnings.warn(
+            f"Unable to load pipeline catalog from {get_url}.\n"
+            f"Will revert to loading backup from {get_path}."
+        )
+        try:
+            catalog = file_utils.load_json(get_path)
```

Comment:
Agree with the temp variable remove!

> Also, I'm wondering if it makes sense to use load_json() as-is here, since if the file is empty, it will throw a JSONDecodeError which will be caught and re-raised by the util but with a misleading error message:

Are you saying that if the file is empty or invalid, the `JSONDecodeError` will **not** be caught by the try-except, and will thus just raise on its own, which might be confusing? Since the path should be in the Error message, I don't think this is super confusing. It should be also pretty uncommon for that to happen to a regular user, as for the file to exist and also be empty/invalid, something must be wrong with our image builds - and even the imprecise error would alert us of that.

Reply:
Hey @surchs, no -- I meant that the current `load_json()` utility func _will_ catch a `JSONDecodeError`, but the error message it currently always prints specifically refers to an invalid "data dictionary" (see code snippet above). This means the error message is inaccurate when the JSON being loaded is e.g., a pipeline catalog JSON, and might confuse users because the `bagel derivatives` command does not take a data dictionary as input.

Since I'm committing to this PR anyways, I'll resolve this here by updating the error message inside `load_json()` to be more generic !